test/test-functions

```diff
@@ -674,7 +674,7 @@ get_ldpath() {
 install_missing_libraries() {
     # install possible missing libraries
     for i in $initdir{,/usr}/{sbin,bin}/* $initdir{,/usr}/lib/systemd/{,tests/{,manual/,unsafe/}}*; do
-        LD_LIBRARY_PATH="${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$(get_ldpath $i)" inst_libs $i
+        LD_LIBRARY_PATH="${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$(get_ldpath $i):$(get_ldpath $i)/src/udev" inst_libs $i
```

Comment:
It's not clear to me how I'd move that up. But I guess this can be done later... (I have the suspicion our build tree should be mostly flat actually for .so and thelike since it is mostly a flat namespace conceptually)

Reply:
Yes, all our build sources and products have unique file names so we don't care about the hierarchy. But meson puts build products in the place where the build rule is defined and is very strict. We jump through some hoops to get the build products in the root build directory. We didn't do that for almost all executables, but not for libraries since we don't refer to directories directly. So it'd be a matter of doing the same thing for libraries, i.e. moving the build root to the meson.build file in project root. Unfortunately that'll make that file even heavier.